discojs/src/training/disco.ts

```diff
@@ -23,56 +28,91 @@ export interface DiscoOptions {
  * communication with nodes, logs and model memory.
  */
 export class Disco {
-  public readonly task: Task
-  public readonly logger: Logger
-  public readonly memory: Memory
-  private readonly client: clients.Client
-  private readonly trainerPromise: Promise<Trainer>
+  readonly #client: clients.Client;
+  readonly #logger: Logger;
 
-  constructor (
+  // small helper to avoid keeping Task & Memory around
+  readonly #updateWorkingModel: (_: Model) => Promise<void>;
+
+  private constructor(
+    public readonly trainer: Trainer,
     task: Task,
-    options: DiscoOptions
+    client: clients.Client,
+    memory: Memory,
+    logger: Logger,
   ) {
-    // Fill undefined options with default values
-    if (options.scheme === undefined) {
-      options.scheme = task.trainingInformation.scheme
-    }
-    if (options.client === undefined) {
-      if (options.url === undefined) {
-        throw new Error('could not determine client from given parameters')
-      }
-      if (options.aggregator === undefined) {
-        options.aggregator = getAggregator(task, options.scheme)
-      }
+    this.#client = client;
+    this.#logger = logger;
+
+    this.#updateWorkingModel = () =>
+      memory.updateWorkingModel(
+        {
+          type: "working",
+          taskID: task.id,
+          name: task.trainingInformation.modelID,
+          tensorBackend: task.trainingInformation.tensorBackend,
+        },
+        this.trainer.model,
+      );
+  }
 
-      if (typeof options.url === 'string') {
-        options.url = new URL(options.url)
+  /**
+   * Connect to the given task and get ready to train.
+   *
+   * Will load the model from memory if available or fetch it from the server.
+   *
+   * @param clientConfig client to connect with or informations on how to create one.
```

Comment:
```suggestion
   * @param clientConfig client to connect with or parameters to create one.
```
information is always singular

Reply:
thanks for being my english Bescherelle :pray: